src/libs/actions/Workflow.ts

```diff
@@ -0,0 +1,196 @@
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {employeeList: previousEmployeeList},
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: null,
+        },
+    ];
+
+    const parameters: CreateWorkspaceApprovalParams = {policyID, authToken, employees: Object.values(employeeList)};
+    API.write(WRITE_COMMANDS.CREATE_WORKSPACE_APPROVAL, parameters, {optimisticData, failureData, successData});
+}
+
+function updateApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWorkflow) {
+    const authToken = NetworkStore.getAuthToken();
+    if (!authToken) {
+        return;
+    }
+
+    const previousEmployeeList = {...allPolicies?.[policyID]?.employeeList};
+    const employeeList = convertApprovalWorkflowToPolicyEmployees({approvalWorkflow, employeeList: previousEmployeeList});
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: {
+                isLoading: true,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {employeeList},
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: {...approvalWorkflow, isLoading: false},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {employeeList: previousEmployeeList},
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: null,
+        },
+    ];
+
+    const parameters: UpdateWorkspaceApprovalParams = {policyID, authToken, employees: Object.values(employeeList)};
+    API.write(WRITE_COMMANDS.UPDATE_WORKSPACE_APPROVAL, parameters, {optimisticData, failureData, successData});
+}
+
+function removeApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWorkflow) {
+    const authToken = NetworkStore.getAuthToken();
+    if (!authToken) {
+        return;
+    }
+
+    const previousEmployeeList = {...allPolicies?.[policyID]?.employeeList};
+    const employeeList = convertApprovalWorkflowToPolicyEmployees({approvalWorkflow, employeeList: previousEmployeeList, removeWorkflow: true});
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: {
+                isLoading: true,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {employeeList},
```

Comment:
There is something else you need to add to this (I'm not sure if you're planning to implement it later or not).

- When the first workflow is added, the `policy.approvalMode` will switch from BASIC to ADVANCED.
- When the last workflow is removed, the `policy.approvalMode` will switch from ADVANCED to BASIC.

Reply:
I knew I forgot about something, thanks. Just updated the logic